src/components/common/SearchInput/SearchInput.tsx

```diff
@@ -40,26 +43,45 @@ const SearchInput = forwardRef<SearchInputRef, SearchInputProps>(
     { onSearchClick, buttonTestId = 'store-search-button', ...props },
     ref
   ) {
+    const [searchQuery, setSearchQuery] = useState<string>('')
+    const [suggestionsOpen, setSuggestionsOpen] = useState<boolean>(false)
+    const searchRef = useRef<HTMLDivElement>(null)
     const { addToSearchHistory } = useSearchHistory()
     const handleSearch = (term: string) => {
       addToSearchHistory(term)
       doSearch(term)
+      setSuggestionsOpen(false)
+      setSearchQuery(term)
     }
 
+    useOnClickOutside(searchRef, () => setSuggestionsOpen(false))
+
     return (
-      <UISearchInput
-        ref={ref}
-        icon={
-          <Icon
-            name="MagnifyingGlass"
-            onClick={onSearchClick}
-            data-testid={buttonTestId}
-          />
-        }
-        placeholder="Search everything at the store"
-        onSubmit={handleSearch}
-        {...props}
-      />
+      <div ref={searchRef} className="search__wrapper">
+        <UISearchInput
+          ref={ref}
+          icon={
+            <Icon
+              name="MagnifyingGlass"
+              onClick={onSearchClick}
+              data-testid={buttonTestId}
+            />
+          }
+          placeholder="Search everything at the store"
+          onChange={(e) => setSearchQuery(e.target.value)}
+          onSubmit={handleSearch}
+          onFocus={() => setSuggestionsOpen(true)}
```

Comment:
wdyt about adding onBlur to setSugggestionOpen to false?

Reply:
why?

Reply:
Sorry, but i couldn't easily test, since the preview isn't working, but I wanna know if the suggestion is hiding if I focus out the search input.